### ISSUE TITLE
reply will not be received when using codec in clustered vertx

### ISSUE THREAD

**Issue by gillbates** [eclipse-vertx/vert.x#5390]
### Questions

Recently we started using cluster for eventbus, and found that the reply cannot be received when reply with POJO.
In the following reproducer, we construct three methods.

1, reply with POJO using local vertx. 
This works fine, which proves our Codec implementation works.

2, reply with String using cluster vertx. In the reproducer, we init two vertx instance to make a cluster.
This works fine, which proves vertx cluster works as expected.

3, reply with POJO using cluster vertx. In the reproducer, we init two vertx instance to make a cluster.
This does NOT work and could be reproduced 100%, maybe some bug there?

### Version

vertx version: 5.1.1
jdk:
```
java version "21.0.1" 2023-10-17 LTS
Java(TM) SE Runtime Environment (build 21.0.1+12-LTS-29)
Java HotSpot(TM) 64-Bit Server VM (build 21.0.1+12-LTS-29, mixed mode, sharing)
```
os: System Version: macOS 15.1 (24B83)

### Do you have a reproducer?

1, start hazelcast with minimal config:
```
docker run -it --rm -e HZ_CLUSTERNAME=hello-world -p 5701:5701 hazelcast/hazelcast:5.5-jdk21
```
2, using the code from https://gist.github.com/gillbates/7be0aae028dd318a1a1393598e898b16

### Steps to reproduce

1. run localVertx(), everything works as expected in local vertx.
2. run clusterVertxReplyWithString(), everything works as expected in hazelcast clustered vertx.
3. run clusterVertxReplyWithPOJO(), something wrong here ...

**Comment by tsegismont** [comment]
Which error do you get?

**Comment by gillbates** [comment]
> Which error do you get?

@tsegismont 
its strange that, no error log is there in console for method clusterVertxReplyWithPOJO ...

**Comment by tsegismont** [comment]
Not even a timeout?

**Comment by gillbates** [comment]
> Not even a timeout?

@tsegismont 

no exceptions or timeout ....

**Comment by tsegismont** [comment]
I took a look at the snippets, its seems the Hazelcast config for the Vert.x node is wrong: it doesn't have any of the required attributes.

Instead of creating it from scratch, you can load the default config and then apply your customizations:

```java
Config hazelcastConfig = ConfigUtil.loadConfig();
// customize hazelcastConfig
ClusterManager mgr = new HazelcastClusterManager(hazelcastConfig);
```

Side note: it's not necessary to start a Hazelcast node upfront unless you want to operate Vert.x nodes [as lite members](https://vertx.io/docs/vertx-hazelcast/java/#_using_lite_members). In this case, you need this:

```java
Config hazelcastConfig = ConfigUtil.loadConfig()
  .setLiteMember(true);
```

**Comment by gillbates** [comment]
> ```java
> Config hazelcastConfig = ConfigUtil.loadConfig();
> ```

@tsegismont thx for the feedback.

but even we use this, the reply cannot be received. Can you provide a minimal code snippet to show that we could receive the POJO?

**Comment by tsegismont** [comment]
I set up a project with the snippets and got an error from Jackson databind: there is no constructor in `SimpleResp` that allows it to create a new instance. You must add this

```java
    public SimpleResp() {
    }
```

**Comment by gillbates** [comment]
> I set up a project with the snippets and got an error from Jackson databind: there is no constructor in `SimpleResp` that allows it to create a new instance. You must add this
> 
> ```java
>     public SimpleResp() {
>     }
> ```


strange, I could run without this error.
Did u manage to receive that reply now besides this?

**Comment by tsegismont** [comment]
Yes

```
[0] registering consumer on /0
[1] registering consumer on /1
[0] sending request to /1
[1] sending request to /0
[1] received reply: Resp{msg='reply from /0', status=1}
[0] received reply: Resp{msg='reply from /1', status=1}
all done ...
```

**Comment by gillbates** [comment]
> Yes
> 
> ```
> [0] registering consumer on /0
> [1] registering consumer on /1
> [0] sending request to /1
> [1] sending request to /0
> [1] received reply: Resp{msg='reply from /0', status=1}
> [0] received reply: Resp{msg='reply from /1', status=1}
> all done ...
> ```

I got it working now after I added following two lines

```
        Config hazelcastConfig = ConfigUtil.loadConfig();
        hazelcastConfig.getNetworkConfig().setPublicAddress("127.0.0.1").setPort(5701).setReuseAddress(true);

        VertxOptions options = new VertxOptions();
        EventBusOptions eventBusOptions = new EventBusOptions();
        eventBusOptions.setHost("0.0.0.0");
        options.setEventBusOptions(eventBusOptions);
```
@tsegismont 
it seemed I must setPublicAddress for hazelcast and setHost for eventBus ...

**Comment by tsegismont** [comment]
I have limited knowledge of Hazelcast configuration, but I don't understand what you're trying to achieve.

There are several nodes started in this program, using the same address and port. Maybe it works incidentally by allowing address reuse, but that looks very complex.

In order to get things working quickly, I've commented out the call to `clusterVertxReplyWithString()` and left only `clusterVertxReplyWithPOJO()`.
And then I used the default config: 

```java
            ClusterManager mgr = new HazelcastClusterManager();
            int finalI = i;
            Future<Void> future = Vertx.builder().withClusterManager(mgr).buildClustered()
```

And this way you don't need any customization of `EventBusOptions` or Hazelcast `Config`.
